develop/tutorials/articles-dxp/310-troubleshooting/13-patching-and-debugging-portal-source-code.markdown

```diff
@@ -0,0 +1,192 @@
+# Patching and Debugging DXP Source [](id=patching-and-debugging-dxp-source)
+
+Auto mechanics, car enthusiasts, and concerned car owners ask "What's under the
+hood?" Reasons for asking could be as follows:
+
+-   Concern about an issue.
+-   Curiosity about the car's capability and inner-workings.
+-   Desire to improve or customize the car.
+
+As a developer, you might have similar reasons for wanting to know "What's under
+the *DXP's* hood?" Since DXP is open-source, you can examine its code and see it
+in action! Setting up DXP with a debugger is your ticket to investigating
+issues, exploring DXP, making improvements and customizations.
+
+Here are the setup steps:
+
+1.   [Download DXP, the DXP source code, and patches](#download-dxp-the-dxp-source-code-and-patches)
+2.   [Prepare DXP](#prepare-dxp)
+3.   [Prepare the DXP source code](#prepare-the-dxp-source-code)
+4.   [Attach a debugger](#debug-dxp)
+
+## Step 1: Download DXP, the DXP source code, and patches [](id=download-dxp-the-dxp-source-code-and-patches)
+
+From the
+[customer portal](https://web.liferay.com/group/customer/dxp),
+download a DXP Digital Enterprise (DXP) bundle (or DXP JARs), and the DXP source
+code for the DXP version you're using. For example, here are ZIP files for a DXP /
+Tomcat bundle and DXP code:
+
+-   `liferay-dxp-digital-enterprise-tomcat-7.0-sp4-20170705142422877.zip`
+-   `liferay-dxp-digital-enterprise-src-7.0-sp4-20170814143552654.zip`
+
+Download fix packs and their source code from
+[here](https://web.liferay.com/group/customer/dxp/downloads/digital-enterprise/fix-packs).
+For example, here are fix pack and source code  ZIP files:
```

Comment:
The -src fix packs contain the full fix pack so if customers would like to use a fix pack, it's enough if they download the -src version.

Reply:
Excellent. Good to know. I'll clarify that.